pure_pursuit_node:
  ros__parameters:
    waypoints_path: "/home/f1tenth/lab_ws25/src/pure_pursuit/racelines/raceline_3.csv"
    odom_topic: "/pf/pose/odom"
    car_refFrame: "laser"
    global_refFrame: "map"
    drive_topic: "/drive"
    rviz_waypointselected_topic: "/waypoint_selected"
    
    # Pure Pursuit Parameters
    # Lookahead is computed using the following: lookahead = min(max(min_lookahead, max_lookahead * curr_velocity / lookahead_ratio), max_lookahead); 
    # min_lookahead: 0.5 # for square track
    min_lookahead: 1.2 # 1.0
    max_lookahead: 3.0 # lookahead when the car is going around max velocity
    lookahead_ratio: 8.0 # approximately the max velocity
    # K_p: 0.25 # for square track
    K_p: 0.25 # for big track
    steering_limit: 25.0
    velocity_percentage: 0.2 # the limit, can only do 1 lap with clean tires


waypoint_visualiser_node:
  ros__parameters:
    waypoints_path: "/home/f1tenth/lab_ws25/src/pure_pursuit/racelines/raceline_3.csv" 
